lib/ruleHelper.js

```diff
@@ -111,7 +111,7 @@ RuleHelper.prototype = {
             break;
         case "MemberExpression":
             methodName = node.property.name;
-            objectName = node.object.name;
+            objectName = node.object.name || this.context.getSource(node.object);
```

Comment:
This makes this.codeName's result look like the following doesn't it:
```
doSomething(blah, thing).write
```

This would make `isAllowedCallExpression` never pass reliably... maybe that is ok?